### ISSUE TITLE
NVDA does not gain proper focus in JetBrains PhpStorm 2018.1.3 x64 when 'Report object shortcut keys' is enabled

### ISSUE THREAD

**Issue by georgizhivankin** [nvaccess/nvda#8322]
### Steps to reproduce:
1. Open JetBrains PhpStorm 2018.1.3 x64
2. Try to access any control in the application or just try to CTRL+TAB or ALT+F.

### Expected behavior:
NVDA should be able to brows through and interact with the controls without requiring the 'Report object shortcut keys' option to be disabled from the 'Object Presentation' tab in Settings in order to allow moving through the controls in the application.

### Actual behavior:
NVDA does not read anything and plays a sound and throws an error in the log instead. It, however, seems to be interacting silently with the said controls.
The error is as follows:
```
IndexError: invalid index
ERROR - eventHandler.executeEvent (11:08:52.378):
error executing event: gainFocus on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionJAB object at 0x07991610> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 152, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\JAB\__init__.pyc", line 527, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 961, in event_gainFocus
  File "NVDAObjects\JAB\__init__.pyc", line 475, in reportFocus
  File "NVDAObjects\__init__.pyc", line 849, in reportFocus
  File "speech.pyc", line 383, in speakObject
  File "speech.pyc", line 275, in speakObjectProperties
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "NVDAObjects\JAB\__init__.pyc", line 243, in _get_keyboardShortcut
IndexError: invalid index
```
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (11:08:54.302):
Developer info for navigator object:
name: u'Editor for pipeline.yml'
role: ROLE_EDITABLETEXT
states: STATE_FOCUSABLE, STATE_EDITABLE, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionJAB object at 0x07991610>
Python class mro: (<class 'NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionJAB'>, <class 'NVDAObjects.behaviors.EditableTextWithoutAutoSelectDetection'>, <class 'editableText.EditableTextWithoutAutoSelectDetection'>, <class 'NVDAObjects.behaviors.EditableText'>, <class 'editableText.EditableText'>, <class 'NVDAObjects.JAB.JAB'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: u''
location: (389, 99, 1125, 10724)
value: None
appModule: <'appModuleHandler' (appName u'phpstorm64', process ID 10592) at address 7750e30>
appModule.productName: u'PhpStorm'
appModule.productVersion: u'2018.1.3.PS-181.4892.97'
TextInfo: <class 'NVDAObjects.JAB.JABTextInfo'>
windowHandle: 10291152
windowClassName: u'SunAwtFrame'
windowControlID: 0
windowStyle: 399441920
windowThreadID: 6856
windowText: u'project-name [C:\\project-name\\pipeline.yml [project-name] - PhpStorm'
displayText: u''`

### System configuration:
NVDA version:
master-15132,354ad355

NVDA Installed

Other information:
Running using JAVA 8 update 171 X64 (system has both X32 and X64 versions installed and JAVA Access Bridge is enabled on both).

Windows version:
Windows 10 Version 1709 (OS Build 16299.125)

Name and version of other software in use when reproducing the issue:

### Other questions:

Does the issue still occur after restarting your PC?
Yes.

Have you tried any other versions of NVDA?
No.

This issue is similar to [NVDA not working with SPSS 24 x64's java access bridge when 'report object shortcut keys' is enabled](https://github.com/nvaccess/nvda/issues/7998)

**Comment by derekriemer** [comment]
Hm. This seems like a bug in JAB, or the implementation.

for index in xrange(bindings.keyBindingsCount):
			binding=bindings.keyBindingInfo[index]
			# We don't support these modifiers
		
However, I think that we could mitigate this.

can you print out the following in the NVDA python console for me?
Go to one of these application items that fails where the error occurs, then do this.
press control+nvda+z.
Paste this.
```
bindings=focus.jabContext.getAccessibleKeyBindings();print "count: %d" % bindings.keyBindingsCount, "Info: %r" % bindings.keyBindingInfo
```
What does it say for a few of these.
You can get at the output with f6.

**Comment by Adriani90** [comment]
@georgizhivankin the discussion is prety old now, could you please give an update on this issue? Have you had success with the bugs reported to them?

**Comment by georgizhivankin** [comment]
@Adriani90 I didn't report more issues sadly but the accessibility of the Jetbrains products is not the best in general and I think they would need much more than filing individual bugs to fix that.

**Comment by mwhapples** [comment]
I think we need an update on whether this is still present. Since the last update there have been Java Access Bridge fixes in NVDA. Also the fact of uninstalling java and reinstalling and enabling access bridge suggests that may be there was some sort of mismatch between the various access bridge DLLs.

**Comment by georgizhivankin** [comment]
@mwhapples I personally can't test it right now as I don't have an active JetBrains license for this year, but if I get one soon, I'll test and report back.

**Comment by Adriani90** [comment]
@georgizhivankin, are you still reproducing this issue with NvDA 2024.2 or 2024.3 beta in PHP storm 2024.2?
The complete GUI of PHP storm seems to be changed in 2024.2, you can test it from here:
https://blog.jetbrains.com/phpstorm/2024/05/phpstorm-2024-1-early-access-program-is-now-open-2/

**Comment by georgizhivankin** [comment]
@Adriani90 I am not using PHPStorm anymore and don't have a license for it as I moved to exclusively coding in Go, but I use Goland and I suppose their GUIs should be pretty similar. I will try the latest Goland version and report back.

**Comment by georgizhivankin** [comment]
@Adriani90 I am not using PhpStorm anymore and don't have a license for it as I moved to exclusively coding in Go, but I use Goland and I suppose their GUIs should be pretty similar. I have not encountered this issue in Goland at all, so I thought this was fixed a while ago. I wish someone who has a PhpStorm license can check this out.

**Comment by seanbudd** [comment]
Closing. Please open a new issue if you are experiencing issues with phpstorm
